extensions/utils/LocationStatusStream.js

```diff
@@ -0,0 +1,180 @@
+     * @param {function} cb callback
+     * @returns {undefined} undefined
+     */
+    _setupMongoClient(cb) {
+        const mongoUrl = constructConnectionString(this._mongoConfig);
+        MongoClient.connect(mongoUrl, {
+                replicaSet: this._mongoConfig.replicaSet,
+                useNewUrlParser: true,
+                useUnifiedTopology: true,
+        }, (err, client) => {
+            if (err) {
+                this._log.error('Could not connect to MongoDB', {
+                    method: 'ServiceStatusManager._setupMongoClient',
+                    error: err.message,
+                });
+                return cb(err);
+            }
+            // connect to metadata DB
+            this._mongoClient = client.db(this._mongoConfig.database, {
+                ignoreUndefined: true,
+            });
+            // creating a collection in case it doesn't exist, this is done as a safety measure
+            // but normally it will be created by backbeat API that gets deployed before the
+            // lifecycle bucket processor.
+            return this._mongoClient.createCollection(locationStatusCollection, err => {
+                // in case the collection already exists, we ignore the error
+                if (err && err.codeName !== 'NamespaceExists') {
+                    this._log.error('Could not create mongo collection', {
+                        collection: locationStatusCollection,
+                        error: err.message,
+                    });
+                    return cb(err);
+                }
+                this._locationStatusColl = this._mongoClient.collection(locationStatusCollection);
+                this._log.info('Connected to MongoDB', {
+                    method: 'ServiceStatusManager._setupMongoClient',
+                });
+                return cb();
+            });
+        });
+    }
+
+    /**
+     * Initialized the list of paused locations
+     * @param {Function} cb callback
+     * @returns {undefined}
+     */
+    _initializeLocationStatuses(cb) {
+        this._locationStatusColl.find({})
+            .toArray((err, locations) => {
+                if (err) {
+                    this._log.error('Could not fetch location statuses from mongo', {
+                        method: 'ServiceStatusManager._initializeLocationStatuses',
+                        error: err.message,
+                    });
+                    return cb(err);
+                }
+                locations.forEach(location => {
+                    const isPaused = location.value[this._serviceName].paused;
+                    if (isPaused) {
+                        this._pauseServiceForLocation(location._id);
+                    }
+                });
+                return cb();
+            });
+    }
+
+    /**
+     * Initializes a change stream on the location status collection
+     * @param {function} cb callback
+     * @returns {undefined}
+     */
+    _setChangeStream(cb) {
```

Comment:
The reason i used callbacks for this and all the other synchronous functions was for cleanness of the code, so that they can fit neatly into the async.series/waterfalls